dev/ledger.md

```diff
@@ -1390,6 +1484,26 @@ point must be discarded and the entire transaction rejected.
   will cause the program exection to fail.
 - Boxes may not be accessed by an app's `ClearStateProgram`.
 
+## Heartbeat Transaction Semantics
+
+ If a heartbeat transaction's $grp$ is empty, and $f < f_{min}$, the
+ transaction fails to execute unless:
+
+   - The _note_ $N$ is empty
+   - The _lease_ $x$ is empty
+   - The _rekey to address_ $\RekeyTo$ is empty
+   - The _heartbeat_address_, $a$, is $online$
+   - The _heartbeat_address_, $a$, $\ie$ flag is true
+   - The _heartbeat_address_, $a$, is _at risk_ of suspension
+
+ An account is _at risk_ of suspension if the current round is between
+ 100-200 modulo 1000, and the blockseed of the most recent round that
+ is 0 modulo 1000 matches $a$ in the first 5 bits.
+
+ If succesful, the `LastHeartbeat` of the specified heartbeat address
+ $a$ is updated to the current round.
```

Comment:
I'm reading this and thinking "So can I submit a heartbeat transaction with a group and a fee set at all?" is that a valid transaction that doesn't perform a heartbeats duties?

Reply:
You can submit a heartbeat with a fee, and if you do, you can use notes, rekey, and leases.  You can also heartbeat when you are not being challenged.  All of that is fine, and such a heartbeat does work - if you want to pay, go ahead.  We just want to constrain things so that heartbeats are only free in the relatively rare cases that you _need_ them, so you can't spam.

Reply:
I understood this to mean that heartbeat transactions can be sent in various configurations, but they only qualify for this free path if these criteria are met. This special path is needed so that a node can send heartbeats as required without having spending keys for the heartbeat account or a balance to pay for transaction fees. 

That said, nothing in the spec appears to preclude the submission of heartbeat transactions in groups, with fees>min, etc.

Reply:
Correct.